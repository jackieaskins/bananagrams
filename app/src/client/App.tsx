import {
  ChakraProvider,
  Spinner,
  Text,
  ThemeConfig,
  ToastProviderProps,
  extendTheme,
} from "@chakra-ui/react";
import { useEffect, useState } from "react";
import { BrowserRouter } from "react-router-dom";
import AppRoutes from "./AppRoutes";
import LocalStorageProvider from "./LocalStorageProvider";
import BraveBrowserDialog from "./dialogs/BraveBrowserDialog";
import ServerDisconnectDialog from "./dialogs/ServerDisconnectDialog";
import CenteredLayout from "./layouts/CenteredLayout";
import NavMenu from "./menus/NavMenu";
import { NavMenuContext } from "./menus/NavMenuContext";
import { socket } from "./socket";

const config: ThemeConfig = {
  initialColorMode: "system",
  useSystemColorMode: true,
};
const theme = extendTheme({ config });

const toastOptions: ToastProviderProps = {
  defaultOptions: {
    position: "top-left",
  },
};

export default function App(): JSX.Element {
  const [renderNavMenu, setRenderNavMenu] = useState(true);
  const [hasConnected, setHasConnected] = useState(socket.connected);

  useEffect(() => {
    function handleConnect() {
      setHasConnected(true);
    }

    socket.on("connect", handleConnect);
    socket.connect();

    return () => {
      socket.off("connect", handleConnect);
    };
  }, []);

  return (
    <LocalStorageProvider>
      <ChakraProvider theme={theme} toastOptions={toastOptions}>
        {hasConnected ? (
          <NavMenuContext.Provider value={[renderNavMenu, setRenderNavMenu]}>
            <BrowserRouter>
              <ServerDisconnectDialog />
              <AppRoutes />
              {renderNavMenu && <NavMenu />}
              <BraveBrowserDialog />
            </BrowserRouter>
          </NavMenuContext.Provider>
        ) : (
          <CenteredLayout>
            <Text textAlign="center">
              <Spinner />
            </Text>
          </CenteredLayout>
        )}
      </ChakraProvider>
    </LocalStorageProvider>
  );
}
