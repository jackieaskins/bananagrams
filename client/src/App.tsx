import { BrowserRouter } from 'react-router-dom';

import Routes from './Routes';
import { SocketProvider } from './socket/SocketContext';

import 'antd/dist/antd.css';

const App: React.FC = () => (
  <SocketProvider>
    <BrowserRouter>
      <Routes />
    </BrowserRouter>
  </SocketProvider>
);

export default App;
