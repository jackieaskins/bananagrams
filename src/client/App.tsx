import {
  ChakraProvider,
  ThemeConfig,
  ToastProviderProps,
  extendTheme,
} from "@chakra-ui/react";
import { useState } from "react";
import { BrowserRouter } from "react-router-dom";
import AppRoutes from "./AppRoutes";
import LocalStorageProvider from "./LocalStorageProvider";
import ServerDisconnectDialog from "./dialogs/ServerDisconnectDialog";
import NavMenu from "./menus/NavMenu";
import { NavMenuContext } from "./menus/NavMenuContext";
import useSocketManager from "./socket/useSocketManager";

const config: ThemeConfig = {
  initialColorMode: "system",
  useSystemColorMode: true,
};
const theme = extendTheme({ config });

const toastOptions: ToastProviderProps = {
  defaultOptions: {
    position: "top-left",
  },
};

export default function App(): JSX.Element {
  useSocketManager();
  const [renderNavMenu, setRenderNavMenu] = useState(true);

  return (
    <LocalStorageProvider>
      <ChakraProvider theme={theme} toastOptions={toastOptions}>
        <NavMenuContext.Provider value={[renderNavMenu, setRenderNavMenu]}>
          <BrowserRouter>
            <ServerDisconnectDialog />
            <AppRoutes />
            {renderNavMenu && <NavMenu />}
          </BrowserRouter>
        </NavMenuContext.Provider>
      </ChakraProvider>
    </LocalStorageProvider>
  );
}
