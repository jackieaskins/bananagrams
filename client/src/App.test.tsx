import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from './App';

describe('App', () => {
  const renderer = ShallowRenderer.createRenderer();

  test('renders properly', () => {
    expect(renderer.render(<App />)).toMatchSnapshot();
  });
});
