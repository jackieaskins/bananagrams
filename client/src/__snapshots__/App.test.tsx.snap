// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders properly 1`] = `
<SocketProvider>
  <BrowserRouter>
    <Routes />
  </BrowserRouter>
</SocketProvider>
`;
